include/drivers/pinctrl/pinctrl_soc_mcux_rt_common.h

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2021, NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_PINCTRL_SOC_MCUX_RT_COMMON_H_
+#define ZEPHYR_INCLUDE_DRIVERS_PINCTRL_SOC_MCUX_RT_COMMON_H_
+
+#include <devicetree.h>
+#include <zephyr/types.h>
+#include <fsl_common.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct pinctrl_soc_pinmux {
+	uint32_t mux_register;
+	uint32_t mux_mode;
+	uint32_t input_register;
+	uint32_t input_daisy;
+	uint32_t config_register;
+	uint32_t input_on_field;
+};
+
+typedef struct pinctrl_soc_pin {
+	struct pinctrl_soc_pinmux pinmux;
+	uint32_t pin_ctrl_flags;
+} pinctrl_soc_pin_t;
+
+#define MCUX_RT_HYS_SHIFT   16
```

Comment:
These constants are defined in the device files in the HAL, see [here](https://raw.githubusercontent.com/zephyrproject-rtos/hal_nxp/master/mcux/mcux-sdk/devices/MIMXRT1062/MIMXRT1062.h)

Reply:
right, I just see those bit pos are common for rt series, so I put it here, and using hal defines would be better.

Reply:
does the DTS use these definitions directly? I thought it pulled from the yaml file for pinctrl, and that these constants would only be used in the C driver file

Reply:
right, dts does not use this, update to use sdk macros.